pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1484 @@
+    pyomo_component,
+    latex_component_map=None,
+    write_object=None,
+    use_equation_environment=False,
+    split_continuous_sets=False,
+    use_short_descriptors=False,
+    fontsize=None,
+    paper_dimensions=None,
+    throw_templatization_error=False,
+):
+    """This function produces a string that can be rendered as LaTeX
+
+    As described, this function produces a string that can be rendered as LaTeX
+
+    Parameters
+    ----------
+    pyomo_component: _BlockData or Model or Objective or Constraint or Expression
+        The Pyomo component to be printed
+
+    latex_component_map: pyomo.common.collections.component_map.ComponentMap
+        A map keyed by Pyomo component, values become the latex representation in
+        the printer
+
+    write_object: io.TextIOWrapper or io.StringIO or str
+        The object to print the latex string to.  Can be an open file object,
+        string I/O object, or a string for a filename to write to
+
+    use_equation_environment: bool
+        If False, the equation/aligned construction is used to create a single
+         LaTeX equation.  If True, then the align environment is used in LaTeX and
+         each constraint and objective will be given an individual equation number
+
+    split_continuous_sets: bool
+        If False, all sums will be done over 'index in set' or similar.  If True,
+        sums will be done over 'i=1' to 'N' or similar if the set is a continuous
+        set
+
+    use_short_descriptors: bool
+        If False, will print full 'minimize' and 'subject to' etc.  If true, uses
+        'min' and 's.t.' instead
+
+    fontsize: str or int
+        Sets the font size of the latex output when writing to a file.  Can take
+        in any of the latex font size keywords ['tiny', 'scriptsize',
+        'footnotesize', 'small', 'normalsize', 'large', 'Large', 'LARGE', huge',
+        'Huge'], or an integer referenced off of 'normalsize' (ex: small is -1,
+        Large is +2)
+
+    paper_dimensions: dict
+        A dictionary that controls the paper margins and size.  Keys are:
+        [ 'height', 'width', 'margin_left', 'margin_right', 'margin_top',
+        'margin_bottom' ].  Default is standard 8.5x11 with one inch margins.
+        Values are in inches
+
+    throw_templatization_error: bool
+        Option to throw an error on templatization failure rather than
+        printing each constraint individually, useful for very large models
+
+
+    Returns
+    -------
+    str
+        A LaTeX string of the pyomo_component
+
+    """
+
+    # Various setup things
+
+    # is Single implies Objective, constraint, or expression
+    # these objects require a slight modification of behavior
+    # isSingle==False means a model or block
+
+    if latex_component_map is None:
+        latex_component_map = ComponentMap()
+        existing_components = ComponentSet([])
+    else:
+        existing_components = ComponentSet(list(latex_component_map.keys()))
```

Comment:
Can't this just be `existing_components = latex_component_map.keys()`?

Reply:
I believe this returns a list, and I need a component set to do equality checking

Reply:
You still don't need the call to `list` then do you?

Reply:
.keys() returns an iterator, but I need a list for the initialization of ComponentSet

Reply:
You can initialize a `ComponentSet` with an iterator.